### PR TITLE
CM - payme redirect

### PR DESCRIPTION
**Pull request by watchtower314**

### CODE REVIEW

File: functions/src/payments/providers/payme/payIn/chargeBuyerToken.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ export const chargeBuyerToken = async (body: IChargeBuyerTokenRequest): Promise<
                 headers: {
                     'Content-type': 'application/json'
                 },
-                ...body
+                ...body,
+                sale_return_url: 'https://google.com'
```
Comment (MeyerPVKernelics) [inline functions/src/payments/providers/payme/payIn/chargeBuyerToken.ts]: maybe redirect to app.common.io

**Review by MeyerPVKernelics — CHANGES_REQUESTED** [review]

**Review by MeyerPVKernelics — APPROVED** [review]

File: functions/src/shared/schemas/billingDetailsValidationSchema.ts [hunk 2]
```diff
@@ -39,4 +39,12 @@ export const billingDetailsValidationSchema = yup.object({
      * Postal / ZIP code of the address.
     */
     postalCode: yup.string().required(),
+
+    /**
+     * ID number of Israli users
+     */
+    ID: yup.string().when("country", {
```
Comment (ariel-dao) [inline functions/src/shared/schemas/billingDetailsValidationSchema.ts]: change ID to socialId

File: functions/src/proposals/business/createJoinRequest.ts [hunk 3]
```diff
@@ -40,6 +42,11 @@ export const createJoinRequest = async (payload: CreateRequestToJoinPayload): Pr
     // Do not let them know if that card exists. It is just 'NotFound' even
     // if it exists, but is not theirs
     throw new NotFoundError(payload.cardId, 'card');
+  }*/
+
+  if (payload.cardId !== payload.proposerId) {
```
Comment (ariel-dao) [inline functions/src/proposals/business/createJoinRequest.ts]: Delete , cardId is unique not userId

File: functions/src/payments/providers/payme/payIn/createBuyerTokenPage.ts [hunk 4]
```diff
@@ -12,7 +12,8 @@ export const createBuyerTokenPage = async (body: ICreateBuyerTokenPagePayload) =
                 headers: {
                     'Content-type': 'application/json'
                 },
-                ...body
+                ...body,
+                sale_return_url: 'https://common.io'
```
Comment (ariel-dao) [inline functions/src/payments/providers/payme/payIn/createBuyerTokenPage.ts]: delete sale return URL we will put loader URL when we get

**Review by ariel-dao — CHANGES_REQUESTED** [review]
Left comments

**Comment by ariel-dao** [comment]
waiting for loader URL before merging this

File: functions/src/payments/providers/payme/payIn/chargeBuyerToken.ts (hunk 1 above)
Comment (ariel-dao) [reply functions/src/payments/providers/payme/payIn/chargeBuyerToken.ts]: redirect here will only affect the webview/iframe not the main window

File: functions/src/payments/providers/payme/payIn/createBuyerTokenPage.ts [hunk 5]
```diff
@@ -12,7 +12,8 @@ export const createBuyerTokenPage = async (body: ICreateBuyerTokenPagePayload) =
                 headers: {
                     'Content-type': 'application/json'
                 },
-                ...body
+                ...body,
+                sale_return_url: 'https://storage.googleapis.com/common-daostack.appspot.com/public_img/loader.jpeg'
```
Comment (watchtower314) [inline functions/src/payments/providers/payme/payIn/createBuyerTokenPage.ts:16]: @ariel-dao this is the url from Alex

**Review by ariel-dao — APPROVED** [review]

**Pull request merged**